src/compose/network-manager.ts

```diff
@@ -65,82 +64,66 @@ export async function remove(network: Network) {
-			network.IPAM.Config[0].Gateway === constants.supervisorNetworkGateway
-		);
-	} catch (e) {
+		const network = await docker.getNetwork(iface).inspect();
+		const result =
+			network.Options['com.docker.network.bridge.name'] === iface &&
+			network.IPAM.Config[0].Subnet === subnet &&
+			network.IPAM.Config[0].Gateway === gateway;
+		return result;
+	} catch (e: unknown) {
 		log.warn(
-			`Failed to read docker configuration of network ${constants.supervisorNetworkInterface}:`,
-			e,
+			`Failed to read docker configuration of network ${iface}:`,
+			(e as Error).message,
 		);
 		return false;
 	}
 }
 
-export function ensureSupervisorNetwork(): Bluebird<void> {
-	const removeIt = () => {
-		return Bluebird.resolve(
-			docker.getNetwork(constants.supervisorNetworkInterface).remove(),
-		).then(() => {
-			return docker.getNetwork(constants.supervisorNetworkInterface).inspect();
-		});
-	};
+export async function ensureSupervisorNetwork(): Promise<void> {
+	try {
+		const net = await docker.getNetwork(iface).inspect();
+		if (
+			net.Options['com.docker.network.bridge.name'] !== iface ||
+			net.IPAM.Config[0].Subnet !== subnet ||
+			net.IPAM.Config[0].Gateway !== gateway
+		) {
+			// Remove network if its configs aren't correct
+			await docker.getNetwork(iface).remove();
+			// This will throw a 404 if network has been removed completely
+			return await docker.getNetwork(iface).inspect();
+		}
+	} catch (e: unknown) {
+		if (!isNotFoundError(e)) {
+			return;
+		}
 
-	return Bluebird.resolve(
-		docker.getNetwork(constants.supervisorNetworkInterface).inspect(),
-	)
-		.then((net) => {
-			if (
-				net.Options['com.docker.network.bridge.name'] !==
-					constants.supervisorNetworkInterface ||
-				net.IPAM.Config[0].Subnet !== constants.supervisorNetworkSubnet ||
-				net.IPAM.Config[0].Gateway !== constants.supervisorNetworkGateway
-			) {
-				return removeIt();
-			} else {
-				return exists(supervisorIfaceSysPath).then((networkExists) => {
-					if (!networkExists) {
-						return removeIt();
-					}
-				});
-			}
-		})
-		.catch(NotFoundError, () => {
-			log.debug(`Creating ${constants.supervisorNetworkInterface} network`);
-			return Bluebird.resolve(
-				docker.createNetwork({
-					Name: constants.supervisorNetworkInterface,
-					Options: {
-						'com.docker.network.bridge.name':
-							constants.supervisorNetworkInterface,
-					},
-					IPAM: {
-						Driver: 'default',
-						Config: [
-							{
-								Subnet: constants.supervisorNetworkSubnet,
-								Gateway: constants.supervisorNetworkGateway,
-							},
-						],
+		log.debug(`Creating ${iface} network`);
+		await docker.createNetwork({
+			Name: iface,
+			Options: {
+				'com.docker.network.bridge.name': iface,
+			},
+			IPAM: {
+				Driver: 'default',
+				Config: [
+					{
+						Subnet: subnet,
+						Gateway: gateway,
 					},
-				}),
-			);
+				],
+			},
+			CheckDuplicate: true,
```

Comment:
docker-compose enables this [by default](https://github.com/docker/compose/pull/9585), we should too, in case this prevents cases of duplicate networks as with https://github.com/balena-os/balena-supervisor/issues/1110 and https://github.com/balena-os/balena-supervisor/issues/590. 

In fact, we should probably make this the default for network creation everywhere in the Supervisor.

Reply:
Oh, TIL! Yeah, we should add it to the `Network.create` defaults